spec/unit/widgets/base_spec.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+module LibyuiClient
+  module Widgets
+    RSpec.describe Base do
+      let(:widget_controller) { instance_double('widget_controller') }
+      let(:widget_filter) { { id: 'foo' } }
+      describe '.Base' do
+        context 'when be initialized' do
+          it 'must create Button subclass with default timeout and interval' do
+            @obj = Button.new(widget_controller, widget_filter)
+            aggregate_failures 'for .Button' do
+              expect(@obj).to be_instance_of(Button)
+              expect(@obj.instance_eval('@timeout', __FILE__, __LINE__)).to equal(5)
```

Comment:
As per our discussion last week, we don't there properties are defined in https://github.com/jknphy/libyui_client/blob/master/lib/libyui_client/widgets/base.rb so let's add single test for it checking defaults and setting other value for it.

Reply:
i break it down to one example checking the defaults and then with alternative timeouts and intervals. I hope it is as expected